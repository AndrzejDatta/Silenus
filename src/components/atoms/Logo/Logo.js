import React from "react";
import PropTypes from "prop-types";
import { StyledLogo } from "components/atoms/Logo/Logo.styles";

const Logo = ({ color, logoSize }) => {
  return <StyledLogo fill={color} logoSize={logoSize} />;
};

Logo.defaultProps = {
  color: "#3AA688",
  logoSize: "small",
};
Logo.propTypes = {
  /** color of logo */
  color: PropTypes.string.isRequired,
  /** size of logo */
  logoSize: PropTypes.string.isRequired,
};

export default Logo;
