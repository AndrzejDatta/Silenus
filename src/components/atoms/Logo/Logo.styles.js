import { ReactComponent as logo } from 'assets/icons/logo.svg';
import styled from 'styled-components';

export const StyledLogo = styled(logo)`
  ${(props) => {
    if (props.size === 'tiny') {
      return `
            width:60px;
            height:60px;
            `;
    } else if (props.size === 'small') {
      return `
            width:150px;
            height:150px;
            `;
    } else if (props.size === 'medium') {
      return `
            width:200px;
            height:200px;
            `;
    } else if (props.size === 'large') {
      return `
            width:250px;
            height:250px;
            `;
    }
  }}
`;
