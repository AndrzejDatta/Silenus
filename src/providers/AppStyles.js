import { ThemeProvider } from 'styled-components';
import { theme } from 'assets/styles/theme';
import { GlobalStyle } from 'assets/styles/GlobalStyles';
export const AppStyles = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
};
