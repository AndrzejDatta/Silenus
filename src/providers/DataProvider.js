import { createContext, useState } from "react";
import { dataPL } from "assets/data/data";

export const dataContext = createContext({
  startPage: {},
  loginPage: {},
  Dashboard: {},
  MojeDoniczki: {},
  DodajDoniczke: {},
  MojeRośliny: {},
  KalendarzOpieki: {},
  KontoUżytkownika: {},
  ZidentyfikujRośline: {},
  PotDetails: {},
  PlantDetails: {},
  Navbar: {},
  images: {},
  navIcons: {},
  potIcons: {},
  categoryIcons: {},
  navIconsColor: [],
  setNavIconColor: () => {},
  changeColor: () => {},
  importAll: () => {},
});

const objectName = [
  "startPage",
  "loginPage",
  "Dashboard",
  "MojeDoniczki",
  "DodajDoniczke",
  "MojeRośliny",
  "KalendarzOpieki",
  "KontoUżytkownika",
  "ZidentyfikujRośline",
  "PotDetails",
  "PlantDetails",
  "Navbar",
];

export const DataProvider = ({ children }) => {
  for (let i = 0; i < dataPL.length; i++) {
    objectName[i] = { ...dataPL[i] };
  }
  const [
    startPage,
    loginPage,
    Dashboard,
    MojeDoniczki,
    DodajDoniczke,
    MojeRośliny,
    KalendarzOpieki,
    KontoUżytkownika,
    ZidentyfikujRośline,
    PotDetails,
    PlantDetails,
    Navbar,
  ] = objectName;

  const [navIconsColor, setNavIconColor] = useState([
    {
      name: "plant",
      src: "potted_plant.svg",
      isClicked: false,
    },
    {
      name: "calendar",
      src: "calendar.svg",
      isClicked: false,
    },
    {
      name: "camera",
      src: "camera.svg",
      isClicked: false,
    },
    {
      name: "user",
      src: "accountCircle.svg",
      isClicked: false,
    },
  ]);

  function importAll(r) {
    let images = {};
    r.keys().map((item, index) => {
      images[item.replace("./", "")] = r(item);
    });
    return images;
  }

  const images = importAll(
    require.context(
      "assets/icons/userAccountOption",
      false,
      /\.(png|jpe?g|svg)$/
    )
  );
  const navIcons = importAll(
    require.context("assets/icons/navIcons", false, /\.(png|jpe?g|svg)$/)
  );

  const potIcons = importAll(
    require.context("assets/icons/potStatus", false, /\.(png|jpe?g|svg)$/)
  );

  const categoryIcons = importAll(
    require.context("assets/icons/categoryIcons", false, /\.(png|jpe?g|svg)$/)
  );

  function changeColor(nameElement) {
    setNavIconColor((prevData) =>
      prevData.map((item) => {
        if (item.name === nameElement && !item.isClicked) {
          return {
            ...item,
            isClicked: true,
          };
        } else {
          return {
            ...item,
            isClicked: false,
          };
        }
      })
    );
  }

  return (
    <dataContext.Provider
      value={{
        startPage,
        loginPage,
        Dashboard,
        DodajDoniczke,
        MojeDoniczki,
        MojeRośliny,
        KalendarzOpieki,
        KontoUżytkownika,
        ZidentyfikujRośline,
        PotDetails,
        PlantDetails,
        Navbar,
        images,
        navIcons,
        potIcons,
        categoryIcons,
        navIconsColor,
        setNavIconColor,
        changeColor,
        importAll,
      }}
    >
      {children}
    </dataContext.Provider>
  );
};

//TODO:improve way
