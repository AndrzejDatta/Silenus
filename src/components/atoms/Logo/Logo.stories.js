import React from "react";
import Logo from "components/atoms/Logo/Logo";
import { storiesName } from "helpers/storiesName";

export default {
  title: `${storiesName.atoms}Logo`,
  component: Logo,
  argTypes: {
    size: {
      options: ["tiny", "small", "medium", "large"],
      control: { type: "radio" },
    },
  },
};

const Template = (args) => <Logo {...args} />;

export const Default = Template.bind({});

Default.args = {
  color: "black",
  size: "small",
};
