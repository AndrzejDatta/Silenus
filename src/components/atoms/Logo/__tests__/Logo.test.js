import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { AppStyles } from "providers/AppStyles";
import { render, screen, rerender } from "@testing-library/react";
import Logo from "../Logo";

describe("Test Logo Component", () => {
  it("Test default style", () => {
    const tree = renderer
      .create(
        <AppStyles>
          <Logo />
        </AppStyles>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("width", "150px");
  });
});
