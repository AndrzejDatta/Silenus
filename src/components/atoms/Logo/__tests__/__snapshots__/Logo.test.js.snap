// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Logo Component Test default style 1`] = `
.c0 {
  width: 150px;
  height: 150px;
}

<svg
  className="c0"
  fill="#3AA688"
  logoSize="small"
>
  logo.svg
</svg>
`;
